content-moderation-evaluation/Evaluating-Amazon-Rekognition-Content-Moderation-Service.ipynb

```diff
@@ -0,0 +1,502 @@
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "## Step 1: Setup Notebook <a id=\"step1\"></a>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "First, let's get the latest installations of our dependencies."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "!pip install --upgrade pip\n",
+    "!pip install boto3 --upgrade\n",
+    "!pip install sagemaker --upgrade"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In order to start, it's necessary to create a bucket where to host evaluation dataset, then set proper values for following variables."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "import itertools\n",
+    "import json\n",
+    "import time\n",
+    "import boto3\n",
+    "import sagemaker\n",
+    "\n",
+    "BUCKET = '<YOUR S3 BUCKET NAME>'          # S3 bucket holds your evaluation dataset\n",
+    "FILE_PREFIX = '<IMAGE PREFIX>'              # The prefix for your evaluation dataset\n",
+    "EXP_NAME = '<JOB PREFIX>'                   # S3 prefix for SageMaker Ground Truth labeling job\n",
+    "INPUT_MANIFEST = '<INPUT FILENAME>'         # Input manifest filename for SageMaker Ground Truth labeling job e.g. input.manifest\n",
+    "OUTPUT_MANIFEST = '<OUTPUT FILENAME>'       # Output manifest filename for SageMaker Ground Truth labeling job e.g. output.manifest"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Make sure the bucket is in the same region as this notebook."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "role = sagemaker.get_execution_role()\n",
+    "region = boto3.session.Session().region_name\n",
+    "s3 = boto3.client(\"s3\")\n",
+    "bucket_region = s3.head_bucket(Bucket=BUCKET)[\"ResponseMetadata\"][\"HTTPHeaders\"][\n",
+    "    \"x-amz-bucket-region\"\n",
+    "]\n",
+    "assert (\n",
+    "    bucket_region == region\n",
+    "), \"You S3 bucket {} and this notebook need to be in the same region.\".format(BUCKET)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Step 2: Use Amazon SageMaker Ground Truth service to assign ground truth moderation labels to the evaluation dataset. <a id=\"step2\"></a>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 1. Create input manifest file for Ground Truth job"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Generate file that contains all images' name in evaluation dataset\n",
+    "objects = s3.list_objects_v2(Bucket=BUCKET, Prefix=FILE_PREFIX)\n",
```

Comment:
```suggestion
    "list_objects_v2 = s3.list_objects_v2(Bucket=BUCKET, Prefix=FILE_PREFIX, StartAfter=FILE_PREFIX)\n",
    "objects = list_objects_v2['Contents']\n",
    "while list_objects_v2['IsTruncated']:
    "    list_objects_v2 = s3.list_objects_v2(Bucket=BUCKET, Prefix=FILE_PREFIX, StartAfter=FILE_PREFIX, ContinuationToken=list_objects_v2['NextContinuationToken'])\n",
    "    objects.extend(list_objects_v2['Contents'])\n",
```

Reply:
Instead of updating the code, I have added comments to notebook with your suggestions. due to the time constraint, I will add this in the next iteration if possible.